sql/sql_class.h

```diff
@@ -1588,20 +1590,29 @@ class Bloom_filter {
 
   typedef std::bitset<SIZE> Bit_set;
 
-  /** The bit set, which is allocated in a MEM_ROOT */
-  Bit_set *bit_set;
+  typedef boost::shared_ptr<Bit_set> Bit_set_ptr;
```

Comment:
@dutow I have a strange feeling that using `shared_ptr` here will case us more race condition troubles despite the fact that it may potentially simplify the code.
Although I don't like this either, for 5.7 I would go with canonical volatile pointer approach.
```suggestion
  typedef Bit_set* Bit_set_ptr;
```

Reply:
And for 8.0, should I keep shared pointers?